### ISSUE TITLE
[sui-core] Narwhal blocks / consensus_db grows forever even without significant activity

### ISSUE THREAD

**Issue by gdanezis** [MystenLabs/sui#2275]
We traced the root cause of the devnet stoppage (two node failures) to the fact that `consensus_db` filled all disk space:
```
root@sui-validator-0:/build# du -h /data/sui_config/consensus_db/89cafd721eeb1b13483d5fb0968500e78103e8146b45edb30d93781d946d0aeb/
9.8G	/data/sui_config/consensus_db/89cafd721eeb1b13483d5fb0968500e78103e8146b45edb30d93781d946d0aeb/
```
It seems that this database is growing for ever even if no transactions are being processed.

**Comment by huitseeker** [comment]
cc @akichidis

**Comment by asonnino** [comment]
Should this issue be on the narwhal repo? Everyone who runs narwhal in production should have the same issue

**Comment by huitseeker** [comment]
@asonnino what's the GC configuration?

**Comment by asonnino** [comment]
What do you mean exactly? The current GC garbage collects 50 rounds in the past (but only memory, no storage). The trick part when garbage collecting storage is that slow nodes may need to catchup from far in the past, and may need to download old certificates to commit them

**Comment by lxfind** [comment]
This may no longer be a problem if we delete the entire consensus db at epoch boundaries. @asonnino what do you think?
A separate (much less priority) issue on DevX is that people who try to run validators locally on their laptop may hit similar issues especially because their disk space will be much more limited. We could however make narwhal/tusk to recognize that we are running a single host consensus and do things differently.

**Comment by asonnino** [comment]
For Sui we will be able to cleanup most of the storage at the end of the epoch (which should solve this issue). But we are working with @kchalkias to solve this problem in a way to also works for external users of Narwhal

**Comment by asonnino** [comment]
This PR should take care of it: https://github.com/MystenLabs/narwhal/pull/663

**Comment by dmitri-perelman** [comment]
Hey guys, is it still relevant? Is @asonnino still working on it / the right PoC?

**Comment by asonnino** [comment]
We need to land this PR: https://github.com/MystenLabs/narwhal/pull/663

**Comment by dmitri-perelman** [comment]
Yea, I just wanted to verify that you're still the right PoC, or we should to hand it over to someone else.

**Comment by asonnino** [comment]
of course, if someone wants to take it over it is fine by me

**Comment by dmitri-perelman** [comment]
Had a quick offline chat: the problem of an idle network consuming disk space is less urgent than other stability improvements required for wave 1 launch.

**Comment by asonnino** [comment]
yes, no need to land the associated PR until we are happy with the stability. Do we have a way to de-prioritize this PR?
